src/main/java/edu/hawaii/its/api/service/GroupingAssignmentService.java

```diff
@@ -41,29 +37,35 @@ public class GroupingAssignmentService {
     public static final Log logger = LogFactory.getLog(GroupingAssignmentService.class);
 
     @Autowired
-    private GrouperApiService grouperApiService;
+    private MemberService memberService;
 
     @Autowired
-    private MemberService memberService;
+    private GrouperApiService grouperApiService;
 
     @Autowired
     private GroupingsService groupingsService;
 
+    /**
+     * A list of grouping paths for all groupings, restricted to admins' use only.
+     */
+    public GroupingPaths groupingPaths(String adminUhIdentifier) {
+        logger.info(String.format("allGroupings; adminUhIdentifier: %s;", adminUhIdentifier));
+        if (!memberService.isAdmin(adminUhIdentifier)) {
+            throw new AccessDeniedException();
+        }
+        return new GroupingPaths(groupingsService.allGroupingPaths());
+    }
 
     /**
-     * Returns an adminLists object containing the list of all admins and all groupings.
+     * Returns groupingsAdmins object containing the list of all admins.
      */
-    public AdminListsHolder adminsGroupings(String adminUhIdentifier) {
-        logger.info(String.format("adminsGroupings; adminUhIdentifier: %s;", adminUhIdentifier));
+    public GroupingAdmins groupingAdmins (String adminUhIdentifier) {
+        logger.info(String.format("groupingAdmins; adminUhIdentifier: %s;", adminUhIdentifier));
         if (!memberService.isAdmin(adminUhIdentifier)) {
             throw new AccessDeniedException();
         }
-        AdminListsHolder adminListsHolder = new AdminListsHolder();
-        List<String> adminGrouping = Arrays.asList(GROUPING_ADMINS);
-        Group admin = getMembers(adminUhIdentifier, adminGrouping).get(GROUPING_ADMINS);
-        adminListsHolder.setAllGroupingPaths(groupingsService.allGroupingPaths());
-        adminListsHolder.setAdminGroup(admin);
-        return adminListsHolder;
+        Group admin = this.getMembers(adminUhIdentifier, Arrays.asList(GROUPING_ADMINS)).get(GROUPING_ADMINS);
+        return new GroupingAdmins(admin);
```

Comment:
I have a better idea for this function, let's change up the `GroupingAdmins` to store a `GroupingGroupMembers` (Note the "s" only after Members) instead of a `Group`. That would mean that `grouperApiService.getMembersResult()` should be called in this function. Making this change will match the code with our current API design.

Reply:
Also could you remove the space before `(string adminUhIdentifier)` on line 62? Thanks!

Reply:
`package edu.hawaii.its.api.groupings;

import edu.hawaii.its.api.wrapper.GetMembersResult;
import edu.hawaii.its.api.wrapper.Subject;

import java.util.List;

public class GroupingAdmins {

    private GroupingGroupMembers adminGroup;

    public GroupingAdmins() {
        this.adminGroup = new GroupingGroupMembers();
        this.adminGroup.setResultCode("FAILURE");
    }

    public GroupingAdmins(GetMembersResult getMembersResult) {
        this.adminGroup = new GroupingGroupMembers(getMembersResult);
        this.adminGroup.setResultCode("SUCCESS");
    }

    public List<GroupingGroupMember> getAdminGroup() { return adminGroup.getMembers(); }

    public void setAdminGroup(List<Subject> adminGroup) { this.adminGroup.setMembers(adminGroup); }

    public String getResultCode() { return this.adminGroup.getResultCode(); }
}
`

Does it look right?

Reply:
Should it have a 3rd constructor taking List<Subjects> as a parameter?

Reply:
The GroupingAdmins should continue to implement `GroupingResult`. The `adminGroupingGroupMembers` can just be `admins` for short.

You only need a 3rd constructor if you will use it.

The getters should be of type `GroupingGroupMembers`. The setters in this case are not needed and should be used as private a function to help with your constructors. You can see the other classes in the /groupings folder that do it like that.